libmamba/src/specs/version.cpp

```diff
@@ -337,6 +366,105 @@ namespace mamba::specs
         return compare_three_way(*this, other) != strong_ordering::less;
     }
 
+    namespace
+    {
+        struct AlwaysEqual
+        {
+        };
+
+        template <typename T>
+        auto starts_with_three_way(const AlwaysEqual&, const T&) -> strong_ordering
+        {
+            return strong_ordering::equal;
+        }
+
+        template <typename T>
+        auto starts_with_three_way(const T&, const AlwaysEqual&) -> strong_ordering
+        {
+            return strong_ordering::equal;
+        }
```

Comment:
There is an ambiguity if someone calls `starts_with_three_way(AlwaysEqual(), AlwaysEqual());` I guess this combination does not make sense, but it might be worth precising it in a comment to avoid the future definition of a useless overload.

Reply:
How about adding
```cpp
[[maybe_unused]] auto starts_with_three_way(const AlwaysEqual&, const AlwaysEqual&) -> strong_ordering {
    static_assert(false, "This comparison should not happen with the current implementation.");
}
```

Reply:
Even better

Reply:
It doesn't work since this is not a template, I made a regular `assert`.